ui/desktop/app/routes/scopes/scope/projects/targets/target/hosts.js

```diff
@@ -0,0 +1,35 @@
+import Route from '@ember/routing/route';
+import { all } from 'rsvp';
+import { A } from '@ember/array';
+
+export default class ScopesScopeProjectsTargetsTargetHostsRoute extends Route {
+  // =methods
+
+  /**
+   * Loads all host-sets and associated hosts under the current target.
+   * @return {Promise{[HostModel]}}
+   */
+  async model() {
+    const { host_sets } = this.modelFor('scopes.scope.projects.targets.target');
+    const hostSetsPromises = await all(
```

Comment:
I think you can call this `hostSets` if you're awaiting.  The resolved value will be host sets, not promises.